File: Exercises_Java/Alarm_System/src/impl/MotionSensor.java
package impl;

import java.util.Random;

/**
 * Created by lucieburgess on 21/01/2017.
 */
public class MotionSensor implements SecuritySensor {

    private String location;
    private String sensorType;
    private Random randomint;

    public MotionSensor(String location) {
        this.location = location;
        this.sensorType = "Motion_sensor";
        randomint = new Random();
        //randomint.setSeed(1); // not sure if I need this or not. Will give a consistent result?
    }

    /**
     * Raises an alarm 20% of the time it is called.
     *
     * @return true 20% of the time
     */
    @Override
    public boolean isTriggered() {
        return randomint.nextInt(100) <= 20;
    }

    @Override
    public String getLocation() {
        return location;
    }

    @Override
    public String getSensorType() {
        return sensorType;
    }

}

File: exercises/week_02/src/main/scala/sensors/MotionSensor.scala
package sensors

/**
  * Created by lucieburgess on 14/05/2017.
  */

import traits.{SecuritySensor, Sensor}

import scala.util.Random

class MotionSensor(private val location: String, private val sensorType: String = "Motion_sensor")
                                      extends Sensor with SecuritySensor {

  override def isTriggered: Boolean = {
    Random.nextInt(100) < 5
  }

  override def getLocation: String = location

  override def getSensorType: String = sensorType



  }
